frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,324 @@
+import React, {useState} from 'react';
+import Modal from "react-bootstrap/Modal";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import ErrorMessage from "../../messages/errorMessage";
+import {AsyncTypeahead} from "react-bootstrap-typeahead";
+import Dropdown from "react-bootstrap/Dropdown";
+import validateWarehouse from "../../validation/WarehouseValidationRules";
+import {FaTrash} from "react-icons/fa";
+
+function ModalAddWarehouse(props) {
+
+    const ref = React.createRef();
+    const [stateOptions, setStateOptions] = useState([]);
+    const [dispatcherOptions, setDispatcherOptions] = useState([]);
+    const [dispatchersList, setDispatchersList] = useState([]);
+    const [dropdownMenuName, setDropdownMenuName] = useState("select type");
+    const [warehouseDto, setWarehouseDto] = useState({
+        id: '',
+        customerId: '',
+        identifier: '',
+        type: '',
+        addressDto: {
+            state: {}
+        },
+        totalCapacity: '',
+        dispatchersId: []
+    });
+
+    const [errors, setErrors] = useState({
+        validationErrors: [],
+        serverErrors: ''
+    });
+
+    const filterByState = () => true;
+    const filterByUsername = () => true;
+
+    const handleStateSearch = (query) => {
+        fetch(`/customers/${props.currentCustomerId}/states?state=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                setStateOptions(res);
+            });
+    };
+
+    const handleDispatcherSearch = (query) => {
+        fetch(`/customers/${props.currentCustomerId}/users/dispatchers?username=${query}`)
+            .then(resp => resp.json())
+            .then(res => {
+                const optionsFromBack = res.map((i) => ({
+                    id: i.id,
+                    name: i.name,
+                    surname: i.surname,
+                    username: i.username
+                }));
+                setDispatcherOptions(optionsFromBack);
+            });
+    };
+
+    const onChangeState = (e) => {
+        e.length > 0 ?
+            setWarehouseDto(preState => ({
+                ...preState,
+                addressDto: {...preState.addressDto, state: {id: e[0].id, state: e[0].state}}
+            })) :
+            setWarehouseDto(preState => ({
+                ...preState,
+                addressDto: {...preState.addressDto, state: {id: '', state: ''}}
+            }));
+    };
```

Comment:
Put e.length > 0 and ternary operator to the place where these to settings are different (state setting.). So you'll have one setWareHouseDto statement.